pkgs/by-name/gh/ghostty/package.nix

```diff
@@ -0,0 +1,149 @@
+{
+  lib,
+  stdenv,
+  bzip2,
+  callPackage,
+  expat,
+  fetchFromGitHub,
+  fontconfig,
+  freetype,
+  git,
+  glib,
+  gsettings-desktop-schemas,
+  gtk4,
+  harfbuzz,
+  libadwaita,
+  libGL,
+  libpng,
+  libX11,
+  libXcursor,
+  libXi,
+  libXrandr,
+  ncurses,
+  nix-update-script,
+  nixosTests,
+  oniguruma,
+  pandoc,
+  pkg-config,
+  removeReferencesTo,
+  versionCheckHook,
+  wrapGAppsHook4,
+  zig_0_13,
+  zlib,
+}:
+let
+  # Ghostty needs to be built with --release=fast, --release=debug and
+  # --release=safe enable too many runtime safety checks.
+  zig_hook = zig_0_13.hook.overrideAttrs {
+    zig_default_flags = "-Dcpu=baseline -Doptimize=ReleaseFast --color off";
+  };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ghostty";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "ghostty-org";
+    repo = "ghostty";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-AHI1Z4mfgXkNwQA8xYq4tS0/BARbHL7gQUT41vCxQTM=";
+  };
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    glib # Required for `glib-compile-schemas`
+    ncurses
+    pandoc
+    pkg-config
+    removeReferencesTo
+    wrapGAppsHook4
+    zig_hook
+  ];
+
+  buildInputs = [
```

Comment:
I think it will be nice when we have conditional buildInputs for Linux. It can be help for supporting MacOS

Reply:
That's something you'd add specifically for porting and, unless there's good reason to, I don't see a point in making every input conditional.